spring-content-autoconfigure/src/main/java/internal/org/springframework/content/fs/boot/autoconfigure/FilesystemContentAutoConfiguration.java

```diff
@@ -21,44 +23,54 @@
 
 @Configuration
 @ConditionalOnClass(FilesystemStoreRegistrar.class)
-@Import({FilesystemContentAutoConfigureRegistrar.class, FilesystemStoreConfiguration.class})
+@Import({ FilesystemContentAutoConfigureRegistrar.class, FilesystemStoreConfiguration.class })
 public class FilesystemContentAutoConfiguration {
 
-    @Autowired
-    private Environment env;
+	@Autowired
+	private Environment env;
 
-
-    @Bean
+	@Bean
 	@ConditionalOnMissingBean(FileSystemResourceLoader.class)
 	FileSystemResourceLoader fileSystemResourceLoader(FilesystemProperties props) {
 		return new FileSystemResourceLoader(props.getFilesystemRoot());
 	}
 
 	@Component
-	@ConfigurationProperties(prefix = "spring.content.fs")
+	@ConfigurationProperties(prefix = "spring.content.fs", ignoreInvalidFields = false, ignoreUnknownFields = false)
 	public static class FilesystemProperties {
 
-	    private static final Logger logger = LoggerFactory.getLogger(FilesystemProperties.class);
+		private static final Logger logger = LoggerFactory.getLogger(FilesystemProperties.class);
 
-	    /**
-	     * The root location where file system stores place their content
-	     */
+		/**
+		 * The root location where file system stores place their content
+		 */
 		String filesystemRoot;
 
 		public String getFilesystemRoot() {
-            if (filesystemRoot == null) {
-                try {
-                    filesystemRoot = Files.createTempDirectory("").toString();
-                } catch (IOException ioe) {
-                    logger.error(String.format("Unexpected error defaulting filesystem root to %s", filesystemRoot), ioe);
-                }
-            }
-
-		    return this.filesystemRoot;
+			if (filesystemRoot == null) {
+				// first try to take system property!!!
```

Comment:
Pretty sure this code isn't required.  This class is marked as a @ConfigurationProperties and therefore Spring Boot will automatically bind filesystemRoot to various system properties, env vars and application properties as outlined [here](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-relaxed-binding).

If that isn't working then we should investigate why rather than try and replace that logic with our own code.

Reply:
Actually this I need saw it better :)